ui/src/app/edge/history/history.component.ts

```diff
@@ -13,44 +13,34 @@ type PeriodString = "today" | "yesterday" | "lastWeek" | "lastMonth" | "lastYear
 })
 export class HistoryComponent implements OnInit {
 
-  private readonly TODAY = new Date();
-  private readonly YESTERDAY = subDays(new Date(), 1);
-  private readonly TOMORROW = addDays(new Date(), 1);
-
-  public activePeriodText: string = "";
   // sets the height for a chart. This is recalculated on every window resize.
   public socChartHeight: string = "250px";
   public energyChartHeight: string = "250px";
-  public activePeriod: PeriodString = "today";
-  public fromDate = this.TODAY;
-  public toDate = this.TODAY;
 
+  // holds the Edge dependend Widget names
+  public widgetNames: string[] = [];
+
+  // holds the current Edge
   protected edge: Edge = null;
-  protected dateRange: IMyDateRange;
-  protected dateRangePickerOptions: IMyDrpOptions = {
-    inline: true,
-    showClearBtn: false,
-    showApplyBtn: false,
-    dateFormat: 'dd.mm.yyyy',
-    disableUntil: { day: 1, month: 1, year: 2013 }, // TODO start with date since the edge is available
-    disableSince: this.toIMyDate(this.TOMORROW),
-    showWeekNumbers: true,
-    showClearDateRangeBtn: false,
-    editableDateRangeField: false,
-    openSelectorOnInputClick: true,
-  };
 
   constructor(
     private route: ActivatedRoute,
-    private translate: TranslateService,
     private service: Service,
-  ) {
-  }
+  ) { }
 
   ngOnInit() {
     this.service.setCurrentComponent('', this.route).then(edge => {
       this.edge = edge;
     });
+    this.service.getWidgets().then(widgets => {
+      let result: string[] = [];
+      for (let widget of widgets) {
+        if (!result.includes(widget.name.toString())) {
+          result.push(widget.name.toString());
+        }
+      }
+      this.widgetNames = result;

```

Comment:
Diese Lösung gefällt mir noch nicht. Du möchtest, dass je Name nur ein Widget angezeigt wird und nicht mehrere... 
z. B. könntest du die Widgets in so einem Objekt anstatt einem Array ablegen: 
{
  "io.openems.edge.controller...": [{
    "id": "ctrl0"
  }]
}